docs/core-design.md

```diff
@@ -0,0 +1,370 @@
+
+As a platform operator I need to connect my on-premise networks directly to my workload Pods. I need to have the ability to represent these networks in my Kubernetes cluster in such a way that I can easily use them in my workloads.
+
+<p align="center">
+  <img src="images/image5.png?raw=true"/>
+</p>
+
+### Story \#6
+
+As a Kubernetes cluster administrator I wish to isolate workloads based on namespaces and network access via assigning different default Network to a Namespace. I do not want the tenants to change their manifests for that purpose. Those workloads should have the same level of Kubernetes functionality: Services, NetworkPolicies, access to Kubernetes API.
+
+<p align="center">
+  <img src="images/image6.png?raw=true"/>
+</p>
+
+## Overview
+
+This design introduces a new object called “PodNetwork”. This object will be a handle representing a network, Pods can connect to. PodNetwork will hold centralized configuration for these endpoints.
+The PodNetwork object will be an input to the DRA driver, for it to provide the “podNetwork” attributed devices.
+Application developers (end users) will leverage DRA’s APIs (ResourceClaim, ResourceClaimTemplate) to reference these devices in a common way.
+
+<p align="center">
+  <img src="images/image7.png?raw=true"/>
+</p>
+
+### Positioning in DRA
+
+To better understand how Multi-Network will function with DRA we want to define 3 main focus areas:
+
+1. Direct DRA device selection \- here are the AI/ML use cases, or simple device handling for which users do not need centralized configuration.
+2. PodNetwork as Cluster-Level construct to enable orchestration of a common set of configurations \- this gives cluster administrators means to easily maintain uniformity. This design focuses on this area.
+3. PodNetwork as a handle to enable Kubernetes networking primitives: Services, NetworkPolicies, etc.
+
+<p align="center">
+  <img src="images/image8.png?raw=true"/>
+</p>
+
+## Detailed design
+
+### PodNetwork CRD
+
+This design defines a PodNetwork object. This will be a CustomResourceDefinition (CRD). PodNetwork’s form is generic and vendor-neutral. This API object is what we wish to use for the future integration with the rest of the Kubernetes ecosystem.
+
+#### Motivation
+
+As of Today, there is no core object that represents the network that the Pod connects to. It represents the administrative resource at the cluster scope for configuration governance for the Pod connection, which provides the cluster administrators centralized control over what is used by the cluster workloads. Furthermore, this will allow workloads to select more than 1 such network configuration to attach to. In such situations, we want to be able to differentiate where specific Kubernetes functionality needs to be applied, e.g. serviceability which uses labels and namespace placement as the identifiers of what workloads to apply to, this object will be another indicator which interfaces of a Pod to use.
```

Comment:
Some of what is described here is covered by DRA API, for example, selecting which network interface will be attached. It would be good to give the motivation on what requirements will drive a different cluster-scoped API or what its function would be.